### PR TITLE
test: Support running e2e tests locally

### PR DESCRIPTION
**Pull request by astefanutti**
This PR makes it possible to run the e2e tests locally, by running:

```bash
 # Start KCP and the KinD clusters
 $ make local-setup
 # Start KCP GLBC
 $ ./bin/ingress-controller -kubeconfig .kcp/admin.kubeconfig -context admin -dns-provider fake -glbc-kubeconfig tmp/kcp-cluster-glbc-control.kubeconfig
 # Run the e2e test suite
 $ make e2e
 ```

Which gives:

```console
$ go build -o bin ./cmd/...
KUBECONFIG="kcp-glbc/.kcp/admin.kubeconfig" CLUSTERS_KUBECONFIG_DIR="kcp-glbc/tmp" \
        go test -timeout 60m -v ./e2e -tags=e2e
=== RUN   TestIngress
--- PASS: TestIngress (192.04s)
PASS
ok      github.com/kuadrant/kcp-glbc/e2e        192.375s
```

### CODE REVIEW

File: utils/local-setup.sh [hunk 1]
```diff
@@ -130,7 +135,7 @@ do
 done
 
 echo "Starting KCP, sending logs to ${KCP_LOG_FILE}"
-${KCP_BIN} start --push-mode --run-controllers --resources-to-sync=secrets --resources-to-sync=deployments --resources-to-sync=services --resources-to-sync=ingresses.networking.k8s.io --auto-publish-apis > ${KCP_LOG_FILE} 2>&1 &
+${KCP_BIN} start --push-mode --discovery-poll-interval 3s --run-controllers --resources-to-sync=secrets,deployments,services,ingresses.networking.k8s.io --auto-publish-apis > ${KCP_LOG_FILE} 2>&1 &
```
Comment (mikenairn) [inline utils/local-setup.sh:138]: What does `discovery-poll-interval 3s` do?

File: utils/local-setup.sh [hunk 2]
```diff
@@ -154,14 +164,13 @@ kubectl apply -f ./config/crd
 
 echo ""
 echo "KCP PID          : ${KCP_PID}"
-echo "Controller 2 PID : ${CONTROLLER_2}"
 echo ""
 echo "The kind k8s clusters have been registered, and KCP is running, now you should run the kcp-ingress"
 echo "example: "
 echo ""
-echo "       ./bin/ingress-controller -kubeconfig .kcp/admin.kubeconfig -glbc-kubeconfig ${TEMP_DIR}/${KCP_GLBC_KUBECONFIG}"
+echo "       ./bin/ingress-controller -kubeconfig .kcp/admin.kubeconfig -context admin -glbc-kubeconfig ${TEMP_DIR}/${KCP_GLBC_KUBECONFIG}"
```
Comment (mikenairn) [inline utils/local-setup.sh:171]: Why do we need `-context admin`?

File: utils/local-setup.sh (hunk 1 above)
Comment (astefanutti) [reply utils/local-setup.sh:138]: That's the frequency the API importer discovers new APIs/CRDs from physical clusters to negotiate them and import them into workspaces.

Comment (astefanutti) [reply utils/local-setup.sh:138]: That helps speeding up the import. It's the value used by the KCP team for their demos.

File: utils/local-setup.sh (hunk 2 above)
Comment (mikenairn) [reply utils/local-setup.sh:171]: Should we add it here? https://github.com/Kuadrant/kcp-glbc/blob/main/.github/workflows/e2e.yaml#L146

**Review by mikenairn — APPROVED** [review]
/lgtm :+1:

File: utils/local-setup.sh (hunk 2 above)
Comment (astefanutti) [reply utils/local-setup.sh:171]: Right, I've just added it.

File: README.md [hunk 3]
```diff
@@ -70,6 +70,23 @@ kcp-cluster-crc   kcp-cluster-crc   True    ["deployments.apps","ingresses.netwo
...
+
+### Testing
+
+The e2e tests can be executed locally by running the following commands:
+
+```bash
+# Start KCP and the KinD clusters
+$ make local-setup
+# Start KCP GLBC
+$ ./bin/ingress-controller -kubeconfig .kcp/admin.kubeconfig -context admin -dns-provider fake -glbc-kubeconfig tmp/kcp-cluster-glbc-control.kubeconfig
```
[the remark this answers is not in the document]
Comment (astefanutti) [reply README.md:83]: Yes, I though the same but was not sure. From my personal experience, I rather start KCP, the KinD clusters once, also the ingress controller once, or a few times, but run the tests multiple times. That's the main reason I've decomposed in these 3 steps. But I agree we could add a target that does it all.

[the remark this answers is not in the document]
Comment (astefanutti) [reply README.md:83]: @jjaferson if you don't mind, I'm inclined to merge this PR as is, and follow up the creation of the `e2e-local` target in a subsequent PR.

[the remark this answers is not in the document]
Comment (jjaferson) [reply README.md:83]: yeah, we can consider the creation of the target later if needed

**Pull request merged**